### ISSUE TITLE
broken UltiSnips

### ISSUE THREAD

**Issue by mathjiajia** [quangnguyen30192/cmp-nvim-ultisnips#53]
After this new PR #50, my UltiSnips is broken for `tex` file: UltiSnips cannot be triggered.

Before:
<img width="1255" alt="Screenshot 2021-12-15 at 6 19 33 PM" src="https://user-images.githubusercontent.com/35106942/146168522-68c9e576-ad8f-48c0-b035-71b1f8139fc6.png">

After:
<img width="379" alt="Screenshot 2021-12-15 at 6 18 33 PM" src="https://user-images.githubusercontent.com/35106942/146168652-1737cd40-6475-4e66-b8fc-2a70d23d2a83.png">

It does not change even if I add `filetype_source = "treesitter"` in the config of `cmp-nvim-ultisnips`.
Could you help me?

#### More information:
Inspired by [Gilles Castel](https://castel.dev/post/lecture-notes-1/), I am using VimTeX and UltiSnips to writing LaTeX documents.
Based on UltiSnips `context` function, I add below to my `tex` snippets.
``` vimL
global !p
def math():
    return vim.eval('vimtex#syntax#in_mathzone()') == '1'
endglobal
```
Clearly, this `math()` function used syntax from VimTeX.
So, in order to get "correct" context, I used `additional_vim_regex_highlighting` for `latex`.
Below is (part of) my configuration of `nvim-treesitter`.
``` lua
highlight = {
enable = true,
additional_vim_regex_highlighting = "latex"
}
```

**Comment by smjonas** [comment]
Oh man, this is not good, sorry for that :/ By the way, do you `nvim_treesitter` installed? I just created a fix that improves things when that plugin is not installed.

**Comment by mathjiajia** [comment]
> Oh man, this is not good, sorry for that :/ By the way, do you have `nvim_treesitter` installed? I just created a fix that improves things when that plugin is not installed. It is possible that the latest commit fixes this for you.
Thanks.
I have installed `nvim-treesitter`.

As a workaround, I used the last commit of `cmp-nvim-ultisnips`.
So, still works.

**Comment by smjonas** [comment]
I could reproduce it and know what the problem is: UltiSnips uses `tex`, treesitter returns `latex`...
Working on a fix.

**Comment by smjonas** [comment]
Hope everything is working again now :)

**Comment by mathjiajia** [comment]
You are so fast 👍.

Thank you. It was fixed now and worked great with PR #50.

**Comment by smjonas** [comment]
Awesome, I am glad everything is working as expected now!😄
